tests/Architecture/ArchitectureTest.php

```diff
@@ -0,0 +1,9 @@
+<?php
```

Comment:
It seems like this file is causing the issue. When I leave it out, it works.
But because that is not an answer, I took the liberty of investigating the root cause.

Long story short:

Stack-trace of second load:

```
  1   vendor/veewee/xml/src/Xml/Encoding/xml_decode.php:20
  2   vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Dependencies/ObjectDependenciesDescription.php:41
  3   vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Dependencies/ObjectDependenciesDescription.php:37
  4   vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Inheritance/ObjectInheritanceDescription.php:33
  5   vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Methods/ObjectMethodsDescription.php:24
  6   vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Properties/ObjectPropertiesDescription.php:26
  7   vendor/pestphp/pest-plugin-arch/src/Objects/ObjectDescription.php:23
  8   vendor/pestphp/pest-plugin-arch/src/Factories/ObjectDescriptionFactory.php:40
  9   vendor/pestphp/pest-plugin-arch/src/Repositories/ObjectsRepository.php:105
  10  vendor/pestphp/pest-plugin-arch/src/Repositories/ObjectsRepository.php:104
  11  vendor/pestphp/pest-plugin-arch/src/Repositories/ObjectsRepository.php:104
  12  vendor/pestphp/pest-plugin-arch/src/Factories/LayerFactory.php:61
  13  vendor/pestphp/pest-plugin-arch/src/Blueprint.php:68
  14  vendor/pestphp/pest-plugin-arch/src/Expectations/ToUse.php:37
  15  vendor/pestphp/pest-plugin-arch/src/SingleArchExpectation.php:156
  16  vendor/pestphp/pest-plugin-arch/src/GroupArchExpectation.php:138
  17  vendor/pestphp/pest-plugin-arch/src/GroupArchExpectation.php:138
  18  vendor/pestphp/pest-plugin-arch/src/GroupArchExpectation.php:129
```

It is caused by one of the internal classes of `ta-tikoma/phpunit-architecture-test` (which is used by the `pest-plugin-arch`.

https://github.com/ta-tikoma/phpunit-architecture-test/blob/e002c84f8b68aac840e187bff6b4792d68d9f529/src/Asserts/Dependencies/ObjectDependenciesDescription.php#L40-L46

Since it is running `enum_exists($functionName)` with default option: `autoload: true`, it will try to resolve it as an enum. Since the fule exists and contains the function, this results in a fatal error.

I was able to solve it by placing the `function_exists` on top.
It might make sense to solve it in that package (or to disable autoloading on the `xxx_exists` methods - but that might break something)

Reply:
FYI Created a PR
https://github.com/veewee/xml/blob/43c8e0ef4248424d5d5d8c955aa86649e35667db/src/bootstrap.php#L171-L174

Reply:
Thanks for the extra info @veewee! For the time being, I've removed the failing tests so the package can start benefiting from the other tests. But if that PR gets merged, I'll take a look at PRing those failing tests back in if they work 😄

Reply:
Hey @veewee! I've just tried adding them back in, but it seems that this one still doesn't quite work:

```php
test('strict types')
    ->expect('Saloon\XmlWrangler')
    ->toUseStrictTypes();
```

It might be something I'm doing wrong on my end though haha!